esphome/components/ethernet/__init__.py

```diff
@@ -89,6 +91,11 @@
     "GPIO17_OUT": ("CLK_OUT", 17),
 }
 
+SPI_INTERFACE_MAP = {
+    "spi2": "SPI2_HOST",
+    "spi3": "SPI3_HOST",
+}
```

Comment:
This should be using the enum system properly instead of using RawExpression later in the code. THe main spi code uses raw because it cannot use enum due to different platforms. Here we only have esp32 with a single enum.

```suggestion
spi_host_device_t = cg.global_ns.enum("spi_host_device_t")

SPI_INTERFACE_MAP = {
    "spi2": spi_host_device_t.SPI2_HOST,
    "spi3": spi_host_device_t.SPI3_HOST,
}
```

Reply:
We also need to validate the interface is available for the current variant

Reply:
I'll try that.

> We also need to validate the interface is available for the current variant
I don't know what you mean.
I am using the choices afforded me by the esp-idf platform.

Reply:
i created and used such a map in code generation. the old one needs to stay for config validation
https://github.com/rtyle/esphome/commit/5aaf15b5d1a9591eb5d8bad82f1936679de3df5b